### PR TITLE
Limited cleanup on EntityFinancialAccount to ensure hooks are called

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
Limited cleanup on EntityFinancialAccount to ensure hooks are called

Before
----------------------------------------
Function is bad

After
----------------------------------------
Function is still bad - but it calls the hooks

Technical Details
----------------------------------------

Comments
----------------------------------------

### CODE REVIEW

File: CRM/Financial/BAO/EntityFinancialAccount.php [hunk 1]
```diff
@@ -45,28 +45,28 @@ public static function retrieve(&$params, &$defaults = [], &$allValues = []) {
...
     }
     if (!empty($ids['entityFinancialAccount'])) {
-      $financialTypeAccount->id = $ids['entityFinancialAccount'];
-      $financialTypeAccount->find(TRUE);
+      $params['id'] = $ids['entityFinancialAccount'];
+      CRM_Core_Error::deprecatedWarning('passing ids is deprecated');
     }
-    $financialTypeAccount->copyValues($params);
+    $financialTypeAccount = parent::writeRecord($params);
     self::validateRelationship($financialTypeAccount);
```
Comment (colemanw) [inline CRM/Financial/BAO/EntityFinancialAccount.php:69]: @eileenmcnaughton it looks like these lines should be switched. The validation should happen before the record is written, right?

Comment (eileenmcnaughton) [reply CRM/Financial/BAO/EntityFinancialAccount.php:69]: oh dang - arg that looks hard to work around with the writeRecord

Comment (colemanw) [reply CRM/Financial/BAO/EntityFinancialAccount.php:69]: I guess you could wrap it in a transaction and `rollback` if an exception is thrown?

**Comment by eileenmcnaughton** [comment]
hmm too hard for now

**Comment by eileenmcnaughton** [comment]
@colemanw were the blown up tests Conformance tests - that relied on it accepting random data?

**Comment by eileenmcnaughton** [comment]
Gonna put this in the too hard basket for now

**Pull request closed without merging**